File: to-do app/to-do-go/src/controller/todomiddleware/middleware.go
package todomiddleware

import (
	"encoding/json"
	"errors"
	"net/http"
	"strconv"

	"renan.com/todo/src/business/messages"
	"renan.com/todo/src/controller/response"
	"renan.com/todo/src/model"
	"renan.com/todo/src/security"
)

const HEADER_USER_ID string = "userID"
const HEADER_USER_STATUS string = "userStatus"
const HEADER_USER_EMAIL string = "userEmail"

// ValidateAuth check if there is a valid JWT Token for a user
func ValidateAuth(next http.HandlerFunc) http.HandlerFunc {
	return func(w http.ResponseWriter, r *http.Request) {
		userID, status, email, _, _, err := security.GetUserData(r)
		if err != nil {
			response.Err(w, http.StatusUnauthorized, err, r)
			return
		}
		if userID == 0 || status == model.STATUS_INACTIVE {
			response.Err(w, http.StatusUnauthorized, errors.New(messages.GetErrorMessageToken()), r)
			return
		}
		r.Header.Add(HEADER_USER_ID, strconv.FormatUint(userID, 10))
		r.Header.Add(HEADER_USER_STATUS, strconv.FormatUint(status, 10))
		r.Header.Add(HEADER_USER_EMAIL, email)
		newUserToken, err := security.RefreshTokenFromRequest(r)
		if err == nil {
			userDTO := response.CreateUserDTOFromJWT(newUserToken)
			if userDTO.ID > 0 {
				jsonUserDTO, err := json.Marshal(userDTO)
				if err == nil {
					w.Header().Set(security.AUTH, string(jsonUserDTO))
				}
			}
		}
		next(w, r)
	}
}

File: to-do-app/to-do-node/src/controller/middleware/middleware.ts
import { NextFunction, Request, Response } from "express"
import token from "../../business/token/token"
import { AppErrorType } from "../../business/errorType"
import msg from "../../business/messages/messages"
import { UserDTO } from "../../business/dto/dto"

const validateAuth = (req: Request, res: Response, next: NextFunction) => {
    const jwt = token.getJwt(req)
    if (!jwt || jwt === null)
        return res.status(AppErrorType.NOT_ALLOWED).json({ errorMessage: msg.getErrorMessageToken() })

    try {
        const userData = token.getUserData(jwt)
        const newUserData = new UserDTO(
            token.refreshToken(jwt),
            token.EXP_TIME,
            userData.userID,
            userData.firstName,
            userData.lastName,
            userData.email,
            userData.status
        )
        res.setHeader(token.AUTH, JSON.stringify(newUserData))
        next();
    } catch (error) {
        return res.status(AppErrorType.NOT_ALLOWED).json({ errorMessage: msg.getErrorMessageToken() })
    }
}

export default {
    validateAuth,
}